internal/controller/ols_app_server_reconciliator.go

```diff
@@ -240,3 +243,38 @@ func (r *OLSConfigReconciler) reconcileService(ctx context.Context, cr *olsv1alp
 	r.logger.Info("OLS service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileLLMSecrets(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	providerCredentials := ""
+	var providerApiToken string
+	for _, provider := range cr.Spec.LLMConfig.Providers {
+		foundSecret := &corev1.Secret{}
+		err := r.Client.Get(ctx, client.ObjectKey{Name: provider.CredentialsSecretRef.Name, Namespace: r.Options.Namespace}, foundSecret)
+		if err != nil {
+			return fmt.Errorf("failed to fetch provider secret: %w", err)
+		}
+		providerApiToken, err = getSecretContent(r.Client, provider.CredentialsSecretRef.Name, r.Options.Namespace, LLMApiTokenFileName)
```

Comment:
The getSecretContent - do we need to get the secret again inside the function. We just found the Secret , we can send just the reference to get the SecretFiled ? @vishnuchalla

Reply:
Agreed. Changed it accordingly.